plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+
+        update_param_to_check = ["tcp_mss_adjustment", "traffic_type", "security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless"
+                                 , "ip_directed_broadcast", "multiple_ip_to_mac_addresses", "supplicant_based_extended_node_onboarding"
+                                 , "group_policy_enforcement_enabled"]
+        vn_name = anycast.get("vn_name")
+        anycast_mapping = self.get_anycast_gateway_mapping(vn_name)
+
+        if vn_name == "INFRA_VN":
+            params_to_remove = ["security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless",
+                                "ip_directed_broadcast", "multiple_ip_to_mac_addresses"]
+            for param in params_to_remove:
+                update_param_to_check.remove(param)
+        else:
+            update_param_to_check.remove("supplicant_based_extended_node_onboarding")
+            update_param_to_check.remove("group_policy_enforcement_enabled")
+
+        if anycast.get("traffic_type") and anycast_details_in_ccc.get("isCriticalPool") is True:
+            update_param_to_check.remove("traffic_type")
+
+        for param in update_param_to_check:
+            if anycast.get(param) is not None:
+                key_in_ccc = anycast_mapping.get(param)
+                if anycast.get(param) != anycast_details_in_ccc.get(key_in_ccc):
+                    msg = (
+                        "Given parameter '{0}' for the given anycast doesnot match so gateway needs update."
+                    ).format(param)
+                    self.log(msg, "INFO")
+                    return True
+
+        return False
+
+    def get_anycast_gateway_update_payload(self, anycast, anycast_details_in_ccc):
+        """
+        Constructs the payload necessary to update the Anycast Gateway configuration in Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            anycast (dict): A dictionary containing the new configuration parameters for the Anycast Gateway.
+            anycast_details_in_ccc (dict): A dictionary containing the current Anycast Gateway configuration details from CCC.
+        Returns:
+            dict: A dictionary containing the payload needed to update the Anycast Gateway configuration.
+        Description:
+            This function constructs an update payload for the Anycast Gateway by combining existing values from
+            `anycast_details_in_ccc` with any new values specified in the `anycast` dictionary. If a parameter exists in
+            `anycast`, it is prioritized over the existing value. If a parameter does not exist in `anycast`, the function
+            retains the current value from `anycast_details_in_ccc`.
+            The resulting payload can be used for updating the Anycast Gateway configuration in the Cisco Catalyst Center.
+        """
+
+        vn_name = anycast_details_in_ccc.get("virtualNetworkName")
```

Comment:
self.log("Constructing update payload for Anycast Gateway in VN: '{0}'.".format(vn_name), "INFO")